lib/ssm_storage/db.rb

```diff
@@ -31,6 +31,23 @@ def constant_exists?
       Object.const_defined? ACTIVE_RECORD_MODEL
     end
 
+    def convert_boolean(value)
+      value = value.downcase unless value.nil?
+      return true if value == 'true'
+      return false if value == 'false'
+      raise 'Not a valid boolean: must be one of true or false'
+    end
+
+    def transform_class(value, type)
+      type = type.downcase unless type.nil?
+      possible_types = ['string', 'integer', 'boolean', 'float']
+      if possible_types.include? type
+        return value.send("to_#{type[0]}") unless type == 'boolean'
+        convert_boolean(value)
```

Comment:
``` ruby 
raise 'Not a valid class: must be one of string, integer, boolean, or float' unless possible_types.include? type
return value.send("to_#{type[0]}") unless type == 'boolean'
convert_boolean(value)
```

Reply:
You can use `type.to_s.downcase` in the `raise` to also handle nil values. I'd also say that if all we care about is the 1st letter to do the conversion, maybe that's all we check? As long as it's `s, i, f` to that conversion, if it's `b` run the boolean.